### PR TITLE
Sign in

### PR DESCRIPTION
**Pull request by srwilliamg**
## Summary

- Added token generation with jsonwebtoken library
- Added endpoint POST /users/sessions
- Added tests for endpoint POST /users/sessions

## Trello Card

https://trello.com/c/eHShImkT/2-sign-in

### CODE REVIEW

File: app/middlewares/validators/users.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+const { users: errorMessages } = require('../../constants/errorMessages');
+
+const emailDefaultValidation = {
+  isEmail: {
+    bail: true,
+    errorMessage: 'Should be an email'
+  },
+  matches: {
+    options: /(\w|-|\.)+@(wolox)\.(\w|\.){2,15}$/i,
+    errorMessage: 'You may only use email addresses from wolox'
```
Comment (blueJayWays) [inline app/middlewares/validators/users.js]: You could use the constant `OnlyWoloxEmail` defined in yout constant file.

File: app/middlewares/validators/users.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }
+};
+
+UsersSchemas.postSessions = {
+  email: {
+    in: ['body'],
+    notEmpty: true,
+    optional: false,
+    ...emailDefaultValidation,
+    errorMessage: 'Email should not be null or empty'
```
Comment (blueJayWays) [inline app/middlewares/validators/users.js]: This message should also be a constant, along with the other ones

**Review by blueJayWays — CHANGES_REQUESTED** [review]
Excellent work. Just a little detail.

File: app/middlewares/validators/users.js (hunk 2 above)
Comment (srwilliamg) [reply app/middlewares/validators/users.js]: done :)

File: app/middlewares/validators/users.js (hunk 1 above)
Comment (srwilliamg) [reply app/middlewares/validators/users.js]: done :)

**Pull request merged**